lib/handlers/formatting.js

```diff
@@ -25,104 +25,86 @@
+        }
+        if (!type) {
+            return logger.warn(`Formatter ${formatter} does not have a formatter class. Skipping...`);
         }
         const versionArg = this.ceProps(`formatter.${formatter}.version`, '--version');
         const versionRe = this.ceProps(`formatter.${formatter}.versionRe`, '.*');
         try {
-            const result = await this.run(exe, [versionArg], {});
+            const result = await exec.execute(exe, [versionArg], {});
             const match = result.stdout.match(versionRe);
-            this.tools[formatter] = {
+            const HandleClass = Object.values(all).find((key) => key.key === type);
+            if (!HandleClass) {
+                return logger.warn(
+                    `Formatter ${formatter} requests formatter class '${type}' which does not exist. Skipping ...`,
+                );
+            }
+            const styleList = this.ceProps(`formatter.${formatter}.styles`);
+            const styles = styleList === '' ? [] : styleList.split(':');
+            this.formatters[formatter] = new HandleClass({
                 exe: exe,
                 version: match ? match[0] : result.stdout,
                 name: this.ceProps(`formatter.${formatter}.name`, exe),
-                styles: this.ceProps(`formatter.${formatter}.styles`, '').split(':'),
-            };
+                styles,
+                type,
+            });
         } catch (err) {
             logger.warn(`Error while fetching tool info for ${exe}:`, {err});
         }
     }
 
-    supportsStyle(tool, style) {
-        return tool.styles.includes(style);
-    }
-
-    validateFormatRequest(req, res) {
-        let requestedTool = this.tools[req.params.tool];
-        if (!requestedTool) {
-            res.status(422); // Unprocessable Entity
-            res.send({
+    async handle(req, res) {
+        const name = req.params.tool;
+        const formatter = this.formatters[name];
+        if (!formatter) {
+            return res.status(422).send({
                 exit: 2,
-                answer: 'Tool not supported',
+                answer: `Unknown format tool '${name}'`,
             });
-            return false;
         }
-        // Only clang supported for now
         if (!req.body || !req.body.source) {
-            res.send({
-                exit: 0,
-                answer: '',
-            });
-            return false;
+            return res.send({exit: 0, answer: ''});
         }
-        // Hardcoded supported clang-format base styles.
-        // Will need a bit of work if we want to support other tools!
-        if (!this.supportsStyle(requestedTool, req.body.base)) {
-            res.status(422); // Unprocessable Entity
-            res.send({
-                exit: 3,
-                answer: 'Base style not supported',
-            });
-            return false;
+        const args = [...formatter.getDefaultArguments()];
+        if (!formatter.isOneTrueStyle()) {
```

Comment:
Due to the way the frontend sends a preferred style (picking from Clang-Format options in the settings pane) this is a workaround to allow rustfmt to work.

The ideal way to go about this is to have the "preferred style" setting per language.